### PR TITLE
DEV: Hash tokens stored from email_tokens

### PR DESCRIPTION
**Pull request by udan11**

### CODE REVIEW

File: app/models/email_token.rb [hunk 1]
```diff
@@ -1,97 +1,107 @@
...
-    SiteSetting.email_token_valid_hours.hours.ago
+  before_validation do
+    self.email = self.email.downcase if self.email
   end
 
-  def self.unconfirmed
-    where(confirmed: false)
+  before_save do
+    if self.scope.blank?
+      STDERR.puts "EmailToken has an empty scope"
```
Comment (ZogStriP) [inline app/models/email_token.rb:35]: Debug leftover?

File: app/models/user.rb [hunk 2]
```diff
@@ -1034,10 +1034,9 @@ def email_confirmed?
     single_sign_on_record&.external_email&.downcase == email
   end
 
-  def activate
-    if email_token = self.email_tokens.active.where(email: self.email).first
-      EmailToken.confirm(email_token.token, skip_reviewable: true)
-    end
+  def activate # TODO: is this duplicated?
```
Comment (ZogStriP) [inline app/models/user.rb:1037]: Is it?

**Review by ZogStriP — APPROVED** [review]
A couple of tiny comments, but otherwise LGTM 👍

File: app/models/user.rb (hunk 2 above)
Comment (udan11) [reply app/models/user.rb]: Yes, this method and `UserAuthenticator#confirm_email` do roughly the same thing in a different way. It's fixed now. :)

File: app/models/email_token.rb (hunk 1 above)
Comment (udan11) [reply app/models/email_token.rb]: Yes and no. I will leave a deprecation notice here to make sure that `scope` is always filled.

Comment (Fhiltyphil88) [reply app/models/email_token.rb]: - [x] The backbone plugins will continue inline I'm sure you can do image

**Pull request merged**